### PR TITLE
Kube OVN addon

### PR DESCRIPTION
**Pull request by AlexsJones**
### Thank you for making MicroK8s better

Please reference the issue this PR is fixing, or provide a description of the problem addressed.

*Also verify you have:*
* [ x] Read the [contributions](https://github.com/ubuntu/microk8s/blob/master/CONTRIBUTING.md) page.
* [x ] Submitted the [CLA form](https://ubuntu.com/legal/contributors/agreement), if you are a first time contributor.

Tested on ARM64/AMD64 Jammy/Focal
Screenshot 2022-04-30 at 07 40 34

### CODE REVIEW

**Review by kwmonroe — APPROVED** [review]

File: addons/kube-ovn/enable [hunk 1]
```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+
+set -e
+
+source $SNAP/actions/common/utils.sh
+VERSION=`cat ../../addons.yaml | grep kube-ovn -A3 | grep version | cut -d ":" -f 2 | tr -d '"' | xargs`
```
Comment (AlexsJones) [inline addons/kube-ovn/enable:6]: @ktsakalozos is this the correct directory pathing? I had an issue once were it seemed to be only a single directory below.

**Comment by AlexsJones** [comment]
Tested on 1.24

```
ubuntu@node1:/var/snap/microk8s/common/addons/core$ microk8s enable kube-ovn
Infer repository core for addon kube-ovn
Fetching version 1.9
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 89120  100 89120    0     0   241k      0 --:--:-- --:--:-- --:--:--  240k
[Step 1/6] Label kube-ovn-master node
node/node2 not labeled
node/node3 not labeled
node/node4 not labeled
node/node1 not labeled
node/node1 labeled
-------------------------------

[Step 2/6] Install OVN components
Install OVN DB in 192.168.1.116,
customresourcedefinition.apiextensions.k8s.io/vpc-nat-gateways.kubeovn.io created
customresourcedefinition.apiextensions.k8s.io/vpcs.kubeovn.io created
customresourcedefinition.apiextensions.k8s.io/ips.kubeovn.io created
customresourcedefinition.apiextensions.k8s.io/subnets.kubeovn.io created
customresourcedefinition.apiextensions.k8s.io/vlans.kubeovn.io created
customresourcedefinition.apiextensions.k8s.io/provider-networks.kubeovn.io created
customresourcedefinition.apiextensions.k8s.io/security-groups.kubeovn.io created
customresourcedefinition.apiextensions.k8s.io/htbqoses.kubeovn.io created
configmap/ovn-config created
serviceaccount/ovn created
clusterrole.rbac.authorization.k8s.io/system:ovn created
clusterrolebinding.rbac.authorization.k8s.io/ovn created
service/ovn-nb created
service/ovn-sb created
service/ovn-northd created
deployment.apps/ovn-central created
daemonset.apps/ovs-ovn created
Waiting for deployment "ovn-central" rollout to finish: 0 out of 1 new replicas have been updated...
Waiting for deployment "ovn-central" rollout to finish: 0 of 1 updated replicas are available...

```
Screenshot 2022-05-31 at 13 11 37

File: addons/kube-ovn/kube-ovn.yaml [hunk 2]
```diff
@@ -0,0 +1,568 @@
...
+            path: /var/snap/microk8s/common/etc/origin/openvswitch
+        - name: host-run-ovs
+          hostPath:
+            path: /run/openvswitch
+        - name: host-run-ovn
+          hostPath:
+            path: /run/ovn
+        - name: cni-conf
+          hostPath:
+            path: /var/snap/microk8s/current/args/cni-network
```
Comment (ktsakalozos) [inline addons/kube-ovn/kube-ovn.yaml:251]: Here we mount in the pod the `/var/..../current/....`. As current is a symlink this translates to `/var/.../<revision>/....`. As soon as the snap refreshes the mounted directory on the host (`/var/.../<revision>/...`) is invalid as there is now a new revision. So after a couple of revisions the snap will not work. This problem in not new, the same issue exists in calico. For this reason when we refresh the snap we redeploy calico [1]. We should be doing something similar for kube-ovn.

A slightly different issue that we need to address is how refreshes happen in multi-node clusters. When a node refreshes it tries to reload the CNI (calico by default). This PR updates the `cni.yaml` on the node the enable command runs. If we are in a multinode cluster and one of the other nodes refreshes the cni of the node that refreshes gets reloaded. This means that nodes that do not have the calico interfaces "patched out" will revive calico. We need to either get kube-ovn work next to calico and/or make the patching of kube-ovn a distributed op.

[1] https://github.com/canonical/microk8s/blob/master/microk8s-resources/wrappers/apiservice-kicker#L114

Comment (neoaggelos) [reply addons/kube-ovn/kube-ovn.yaml:251]: Addressed in https://github.com/canonical/microk8s/pull/3307

**Pull request merged**